src/components/Card/Card.tsx

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import {
+  addCardNumberDashes,
+  addSeparator,
+  replaceCardNumberToO,
+} from '@/utils/formatter';
+import { CardContainer } from './Card.style';
+import { CardField } from '@/types';
+import { Colors, colors } from '@/styles/colors';
+
+type CardProps = {
+  size: 'small' | 'big';
+  cardColor: Colors;
+  cardName?: string;
+} & Omit<CardField, 'cardPassword' | 'cvc'>;
+
+const Card = ({
+  size,
+  cardNumber,
+  expirationMonth,
+  expirationYear,
+  cardName,
+  cardColor,
+  ownerName,
+}: CardProps) => {
```

Comment:
CardProps를 보시면 스타일에 관한 prop과 card도메인에 관한 prop이 있어요.
여기서 card쪽 prop을 보면 렌더하는데 필요한 데이터가 많은데요. 요걸 객체형식으로 묶어서 관리해보시면 어떨까 싶어요. 그러면  관련있는 데이터가 하나의 객체에 모이게되면서 prop부도 깔끔해지는 효과가 있습니다.

```suggestion
type CardProps = {
  size: 'small' | 'big';
  cardColor: Colors;
  card: Card
} 

const Card~~ = ({  size, color,  card }: CardProps) => {
```

Reply:
React.memo 를 사용해서 카드의 컴포넌트를 메모이제이션이 되려면 원시값으로 props를 넘겨주는게 좋지 않을까 해서 만든 구조입니다!

Reply:
렌더링 최적화를 고려해주셨군요! 저는 그렇다면 객체 레퍼런스를 유지하는게 어떨까 싶어요. 렌더링 최적화 때문에 prop을 원시값으로 받아야한다면 도메인데이터나 컴포넌트 prop 복잡도 관리에 제약이 생기지않을까 생각해봅니다

Reply:
리액트에서 "객체 레퍼런스를 유지" 한다는건 ref 를 사용한다고 이해하면 될까요?

Reply:
렌더링에 관여를 하지 않는 데이터라면 ref를 사용해볼 법도 하지만 대개의 일반적인경우에는  useMemo로 메모이제이션을 해줄 것 같습니다.